packages/@lwc/synthetic-shadow/src/shared/handled-events.ts

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+const currentlyHandledEvents: Set<Event> = new Set();
+
+export function isBeingHandledByWrappedListener(event: Event) {
+    return currentlyHandledEvents.has(event);
+}
+
+export function invokeWrappedListener(listener: EventListener, event: Event) {
+    currentlyHandledEvents.add(event);
+    try {
+        listener(event);
+    } finally {
+        currentlyHandledEvents.delete(event);
+    }
+}
```

Comment:
I would recommend making `invokeWrappedListener` accept a `target` as an argument. This avoids the function creation overhead when doing `listener.bind(target)` prior to invoking `invokeWrappedListener`. 

```suggestion
export function invokeWrappedListener(listener: EventListener, target: EventTarget, event: Event) {
    currentlyHandledEvents.add(event);
    try {
        listener.call(target, event);
    } finally {
        currentlyHandledEvents.delete(event);
    }
}
```